gcloud/storage/_implicit_environ.py

```diff
@@ -81,29 +29,11 @@ class _DefaultsContainer(object):
     :param connection: Persistent implied connection from environment.
     """
 
-    @_lazy_property_deco
-    @staticmethod
-    def project():
-        """Return the implicit default project."""
-        return _determine_default_project()
-
-    def __init__(self, project=None, bucket=None, connection=None,
-                 implicit=False):
-        if project is not None or not implicit:
-            self.project = project
+    def __init__(self, bucket=None, connection=None):
```

Comment:
> What if they want to use a different default project for storage than the one used for pubsub?

What percentage of all the users who ever download `gcloud` would want to do that?  And if they did, wouldn't having to supply the project ID explicitly for at least one of the two be a reasonable thing for us to require?

@jgeewax can you weigh in?